Getting partition location mismatch error for HIVE ACID in newer version(410) of trino

we are getting following error while querying in trino v410. this query works fine on v387. 


```

io.trino.spi.TrinoException: Partition location in OriginalFile 'abfs://clipped.blob.location/entity/extraction_bifrost/1/partition=2022-11-03/partition=2022-11-03/_tmp_b5bb098f-0f06-4236-8cfa-64f1846d1c80/base_0007044' does not match stored location 'abfs://clipped.blob.location/entity/extraction_bifrost/1/partition=2022-11-03/partition=2022-11-03' 	at io.trino.plugin.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:312) 	at io.trino.plugin.hive.util.ResumableTasks$1.run(ResumableTasks.java:38) 	at io.trino.$gen.Trino_410____20230321_092729_2.run(Unknown Source) 	at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:80) 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) 	at java.base/java.lang.Thread.run(Thread.java:833) Caused by: java.lang.IllegalArgumentException: Partition location in OriginalFile 'abfs://clipped.blob.location/entity/extraction_bifrost/1/partition=2022-11-03/partition=2022-11-03/_tmp_b5bb098f-0f06-4236-8cfa-64f1846d1c80/base_0007044' does not match stored location 'abfs://clipped.blob.location/entity/extraction_bifrost/1/partition=2022-11-03/partition=2022-11-03/' 	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:435) 	at io.trino.plugin.hive.AcidInfo$Builder.addOriginalFile(AcidInfo.java:236) 	at io.trino.plugin.hive.BackgroundHiveSplitLoader.getTransactionalSplits(BackgroundHiveSplitLoader.java:723) 	at io.trino.plugin.hive.BackgroundHiveSplitLoader.loadPartition(BackgroundHiveSplitLoader.java:541) 	at io.trino.plugin.hive.BackgroundHiveSplitLoader.loadSplits(BackgroundHiveSplitLoader.java:391) 	at io.trino.plugin.hive.BackgroundHiveSplitLoader$HiveSplitLoaderTask.process(BackgroundHiveSplitLoader.java:305) 	... 6 more

--
```

please help us in solving the issue we are using HIVE

--- a/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/AcidTables.java
+++ b/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/AcidTables.java
@@ -312,8 +312,7 @@ public final class AcidTables
         FileIterator iterator = fileSystem.listFiles(directory);
         while (iterator.hasNext()) {
             FileEntry file = iterator.next();
-            String name = new Path(file.location()).getName();
-            if (!name.startsWith("_") && !name.startsWith(".")) {
+            if (!file.location().contains("/_") && !file.location().contains("/.")) {
                 files.add(file);
             }
         }

--- a/plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/TestAcidTables.java
+++ b/plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/TestAcidTables.java
@@ -117,6 +117,8 @@ public class TestAcidTables
                 new MockFile("mock:/tbl/part1/000002_0", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/random", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/_done", 0, FAKE_DATA),
+                new MockFile("mock:/tbl/part1/_tmp/000000_0", 0, FAKE_DATA),
+                new MockFile("mock:/tbl/part1/_tmp/abc/000000_0", 0, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/subdir/000000_0", 0, FAKE_DATA));
         AcidState state = getAcidState(
                 testingTrinoFileSystem(fs),
@@ -147,6 +149,8 @@ public class TestAcidTables
                 new MockFile("mock:/tbl/part1/000002_0", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/random", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/_done", 0, FAKE_DATA),
+                new MockFile("mock:/tbl/part1/_tmp/000000_0", 0, FAKE_DATA),
+                new MockFile("mock:/tbl/part1/_tmp/delta_025_025/000000_0", 0, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/subdir/000000_0", 0, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/delta_025_025/bucket_0", 0, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/delta_029_029/bucket_0", 0, FAKE_DATA),
@@ -185,6 +189,8 @@ public class TestAcidTables
             throws Exception
     {
         MockFileSystem fs = new MockFileSystem(newEmptyConfiguration(),
+                new MockFile("mock:/tbl/part1/_tmp/bucket_0", 0, FAKE_DATA),
+                new MockFile("mock:/tbl/part1/_tmp/base_5/bucket_0", 0, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/base_5/bucket_0", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/base_10/bucket_0", 500, FAKE_DATA),
                 new MockFile("mock:/tbl/part1/base_49/bucket_0", 500, FAKE_DATA),

--- a/plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/TestHiveAcidUtils.java
+++ b/plugin/trino-hive/src/test/java/io/trino/plugin/hive/util/TestHiveAcidUtils.java
@@ -53,6 +53,8 @@ public class TestHiveAcidUtils
                 new MockFile("mock:/tbl/part1/000002_0", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/random", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/_done", 0, new byte[0]),
+                new MockFile("mock:/tbl/part1/_tmp/000000_0", 0, new byte[0]),
+                new MockFile("mock:/tbl/part1/_tmp/abc/000000_0", 0, new byte[0]),
                 new MockFile("mock:/tbl/part1/subdir/000000_0", 0, new byte[0]));
         AcidUtils.Directory dir = AcidUtils.getAcidState(
                 new MockPath(fs, "/tbl/part1"),
@@ -83,6 +85,8 @@ public class TestHiveAcidUtils
                 new MockFile("mock:/tbl/part1/000002_0", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/random", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/_done", 0, new byte[0]),
+                new MockFile("mock:/tbl/part1/_tmp/000000_0", 0, new byte[0]),
+                new MockFile("mock:/tbl/part1/_tmp/delta_025_025/000000_0", 0, new byte[0]),
                 new MockFile("mock:/tbl/part1/subdir/000000_0", 0, new byte[0]),
                 new MockFile("mock:/tbl/part1/delta_025_025/bucket_0", 0, new byte[0]),
                 new MockFile("mock:/tbl/part1/delta_029_029/bucket_0", 0, new byte[0]),
@@ -123,6 +127,8 @@ public class TestHiveAcidUtils
     {
         Configuration conf = newEmptyConfiguration();
         MockFileSystem fs = new MockFileSystem(conf,
+                new MockFile("mock:/tbl/part1/_tmp/bucket_0", 0, new byte[0]),
+                new MockFile("mock:/tbl/part1/_tmp/base_5/bucket_0", 0, new byte[0]),
                 new MockFile("mock:/tbl/part1/base_5/bucket_0", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/base_10/bucket_0", 500, new byte[0]),
                 new MockFile("mock:/tbl/part1/base_49/bucket_0", 500, new byte[0]),

